### PR TITLE
Updates approximate and smooth, and adds refine and fair, extends validate.

### PR DESCRIPTION
**Pull request by pentacular**
Note that refine is problematic due to non-coplanar edge-flips.

### CODE REVIEW

File: algorithm/cgal/fair.js [hunk 1]
```diff
@@ -3,9 +3,23 @@ import { fromCgalGeometry, withCgalGeometry } from './cgalGeometry.js';
 
 import { ErrorZeroThickness } from './error.js';
 
-export const fair = (inputs, count, density = 0) =>
+export const fair = (
```
Comment (BarbourSmith) [inline algorithm/cgal/fair.js:6]: To what extent are the default parameters OK? I worry a little bit about folks getting started being intimidated by tuning things

File: nb/api/iron.md [hunk 2]
```diff
@@ -0,0 +1,18 @@
+[index](../../nb/api/index.md)
+### iron()
```
Comment (BarbourSmith) [inline nb/api/iron.md:2]: This is my favorite new feature in a while

**Review by BarbourSmith — APPROVED** [review]
Looks awesome to me! 👍 👍 


This one is a super neat example: 

image

File: algorithm/cgal/fair.js (hunk 1 above)
Comment (pentacular) [reply algorithm/cgal/fair.js:6]: Yeah, that'll take a little experimentation to figure out -- but hopefully there should be only one principle knob.

**Pull request merged**